core/arch/riscv/include/kernel/thread_arch.h

```diff
@@ -0,0 +1,142 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#ifndef __KERNEL_THREAD_ARCH_H
+#define __KERNEL_THREAD_ARCH_H
+
+#ifndef __ASSEMBLER__
+#include <compiler.h>
+#include <riscv.h>
+#include <types_ext.h>
+#endif
+
+#ifndef __ASSEMBLER__
+
+#define THREAD_CORE_LOCAL_ALIGNED __aligned(2 * RISCV_XLEN_BYTES)
+
+struct thread_pauth_keys {
```

Comment:
Is this used? I see that you add it in `struct thread_core_local` below, but is it used from there?

Reply:
`thread_resume() `is not part of this commitset yet. But referenced [here ](https://github.com/OP-TEE/optee_os/pull/5606/commits/73ca858cdfd469ed36deb81f771469b2a36f2515#diff-bcc8474c5ac865bd696bc2b4329c46f8616c5d60ad1a712f94def838e291f8afR135).

Reply:
Will that ever be used? I find it somewhat unlikely that pointer authentication is implemented in the same way in Riscv as in Arm.

Reply:
Better to remove them for now.